nautobot/docs/models/extras/relationship.md

```diff
@@ -0,0 +1,79 @@
+# Relationships
+Sometimes it is desirable to create a new kind of relationship between one (or more) objects in your source of truth to reflect business logic or other relationships that may be useful to you but that haven't been defined. This is where the Relationships feature comes in: like defining custom fields to hold atributes specific to your use cases, relationships define specific links between objects that might be specific to your network or data.
+
+## Relationship Types
+
+* Many to Many -  where both sides of the relationship connection can be connected to multiple objects. e.g. VLANs can be connected to multiple devices and devices will have multiple VLANs.
+* One to Many - where one side of the connection can only have one object. e.g. where a controller has many supplicants like FEX and parent switch. A FEX can be uplinked to one parent switch (in most cases), but the parent switch can have many FEX. 
```

Comment:
Currently it's not possible to create a relationship between 2 objects of the same type, so this example "FEX to parent switch" is currently not supported. It was supported in my initial implementation but I removed it at the end until we can implement that properly.